src/library/scala/collection/mutable/PriorityQueue.scala

```diff
@@ -328,6 +328,16 @@ sealed class PriorityQueue[A](implicit val ord: Ordering[A])
     pq
   }
 
+  override def copyToArray[B >: A](xs: Array[B], start: Int): Int = copyToArray(xs, start, length)
```

Comment:
I can try it out, but I'm not sure I follow. Are you saying that accessing the field `length` may be costing?

Reply:
There's a completely separate implementation in `IterableOnceOps`. If it delegated to the version with the length parameter you wouldn't have to override it here. Implementing it as `copyToArray(xs, start, length)` is obviously a bad idea in `IterableOnceOps` (because `length` could be expensive or even even exhaust the `IterableOnce`) but `copyToArray(xs, start, Int.MaxValue)` would be a safe implementation.